elements/html-template-element.es

```diff
@@ -1,51 +1,51 @@
 // INTERESTING! Converting `Template` to a class increases size by ~16 octets
 
-const HTMLTemplateElement = Template = function (name) {
+const Template = HTMLTemplateElement = function (name) {
 
   // create shallow clone using `.getOwnPropertyDescriptors`
   // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptors#Examples
   // https://docs.microsoft.com/en-us/scripting/javascript/reference/object-getownpropertydescriptor-function-javascript
+  // NO IE SUPPORT!!!!
   return Object.assign
     (document.querySelector ('template[name='+name+']'), { bind } )
 
   function bind (context) {
 
-    // https://dom.spec.whatwg.org/#converting-nodes-into-a-node
-    contexts = [].concat ( ... [context] )
-
     let
-      clone
-    , template = this.cloneNode (false)
+      html
+    , template = this.innerHTML
+    , contexts = [].concat ( ... [context] )   // https://dom.spec.whatwg.org/#converting-nodes-into-a-node
+    , keys     = Object.keys (contexts [0])    // memoize keys
+    , tokens   = keys.map (key => '{'+key+'}') // memoize tokens
+    , fragment = document.createElement ('template')
 
-    template.innerHTML =
-    contexts
-      .map (context => context)
-      .map ((context, index) => {
+    , deposit = (context, index,) => {
```

Comment:
trailing comma. amazed this works. i think is the secret behind `...params` "rest params" https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters